src/main/java/teammates/common/util/ActivityLogGenerator.java

```diff
@@ -0,0 +1,336 @@
+package teammates.common.util;
+
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.TimeZone;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.appengine.api.log.AppLogLine;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
```

Comment:
This sounds really familiar. Correct me if I'm wrong, but isn't this dependency what #6442 is about? Is it possible to introduce the same idea here, i.e. just inject the `UserType` parameter from the call sites?